website/source/upgrade-guides/0-7.html.markdown

```diff
@@ -0,0 +1,204 @@
+
+## Maps in Displayed Plans
+
+When displaying a plan, Terraform now distinguishes attributes of type map by using a `%` character for the "length field".
+
+Here is an example showing a diff that includes both a list and a map:
+
+```
+somelist.#:  "0" => "1"
+somelist.0:  "" => "someitem"
+somemap.%:   "0" => "1"
+somemap.foo: "" => "bar"
+```
+
+## Interpolation Changes
+
+There are a few changes to Terraform's interpolation language that may require updates to your configs.
+
+### String Concatenation
+
+The `concat()` interpolation function used to work for both lists and strings. It now only works for lists.
+
+```
+"${concat(var.foo, "-suffix")}"     # => Error! No longer supported.
+```
+
+Instead, you can use nested interpolation braces for string concatenation.
+
+```
+"${"${var.foo}-suffix"}"
+```
+
+### Nested Quotes and Escaping
+
+Escaped quotes inside of interpolations were supported to retain backwards compatibility with older versions of Terraform that allowed them.
+
+Now, escaped quotes will no longer work in the interpolation context:
+
+```
+"${lookup(var.somemap, \"somekey\")}"      # => Syntax Error!
+```
+
+Instead, treat each set of interpolation braces (`${}`) as a new quoting context:
+
+```
+"${lookup(var.somemap, "somekey")}"
+```
+
+This allows double quote characters to be expressed properly within strings inside of interpolation expressions:
+
+```
+"${upper("\"quoted\"")}"    # => "QUOTED"
+```
+
+## Safer `terraform plan` Behavior
+
+Prior to v0.7, the `terraform plan` command had the potential to write updates to the state if changes were detected during the Refresh step (which happens by default during `plan`). Some configurations have metadata that changes which every read, so Refresh would always result in changes to the state, and therefore a write.
+
+In collaborative enviroments with shared remote state, this potential side effect of `plan` would cause unnecessary contention over the state, and potentially even interfere with active `apply` operations if they were happening simultaneously elsewhere.
+
+Terraform v0.7 addresses this by changing the Refresh process that is run during `terraform plan` to always be an in-memory only refresh. New state information detected during this step will not be persisted to permanent state storage.
+
+If the `-out` flag is used to produce a Plan File, the updated state information _will_ be encoded into that file, so that the resulting `terraform apply` operation can detect if any changes occurred that might invalidate the plan.
+
+For most users, this change will not affect your day-to-day usage of Terraform. For users with automation that relies on the old side effect of `plan`, you can use the `terraform refresh` command, which will still persist any changes it discovers.
+
+## Migrating to Data Sources
+
+With the addition of [Data Sources](/docs/configuration/data-sources.html), there are several resources that were acting as Data Sources that are now deprecated.
```

Comment:
@rowleyaj They will print a deprecation warning in 0.7 if you use them a a resource rather than a data source.